docs/server-side.md

```diff
@@ -0,0 +1,306 @@
+# Using Guardian JS Client from server side
+
+You can use Guardian Client from server side, by default Guardian Client from server side
+works exactly like in the browser. Meaning, it opens a web socket connection through socket.io
+and uses it to interact with the server exactly as in the client side.
+
+However, for certain use cases it is better not to have this automatic behavior, in particular,
+you might want to manually check the state of the transaction and avoid not necessary external
+connections or use a polling approach instead of the classic socket-based approach.
+
+## Using manual transaction state check instead of websocket or polling
+
+To manually check the state of the transaction you need to configure guardian JS
+not to use a socket client, you can do so by setting `stateCheckingMechanism` to `manual` in
+the configuration.
+
+Keep in mind, that by doing so you will stop getting automatic events, i.e. you won't receive
+`enrollment-complete` and `auth-response` events and their handlers won't be executed automatically
+when the transaction state changes on Guardian servers.
+
+Enrollment is the only valid scenario for server side usage.
+
+### Serializing a transaction
+Serializing a transaction allows you to save an active transaction and resume it later if it is not
+expired, it is specially suitable for server-side usage were you might want to save a transaction
+once the request finishes and resume it in a follow up request.
+
+The following method will create a new transaction if a ticket is available or try to resume it
+from session.
+
+```js
+function getGuardianTransaction(req, options, cb) {
+  options = options || {};
+
+  if (options.ticket) {
+    // Start a new transaction if a ticket is available
+
+    return void guardianJSBuilder({
+      serviceUrl: process.env.GUARDIAN_URL,
+      ticket: options.ticket,
+
+      issuer: {
+        label: env.GUARDIAN_ISSUER_LABEL,
+        name: env.GUARDIAN_ISSUER_NAME
+      },
+
+      accountLabel: req.user.email,
+
+      stateCheckingMechanism: 'manual'
+    }).start(cb);
+  }
+
+  if (req.session.mfaTx) {
+    // Resume a transaction if it is available in session
+
+    return void guardianJSBuilder.resume({ stateCheckingMechanism: 'manual' }, req.session.mfaTx, cb);
+  }
+
+  cb(new HttpError(403, 'no_session_active', 'There is no valid mfa transaction active, nor ticket provided'));
+}
+```
+
+### Enrollment example 1: supports all methods using manual state checking mechanism
+
+```js
+// WARNING 1: This is an advanced example, before using it consider if the
+// hosted-page option does not match you use case, they are easier to implement
+// and better suitted for most use cases.
+//
+// WARNING 2: POST Methods require CSRF protection, don't forget to add them or
+// you will be at risk of CSRF attacks, this example left them out for simplicity
+// reasons and because there are many different ways to add them that
+// are really specific to your enrollment.
```

Comment:
I think this warning is superfluous; the default for a browser is to block XHR posts to other domains, so anyone adding a header for allowed origins do so at their own risk.